src/types/widget.ts

```diff
@@ -7,41 +7,41 @@ import {
 } from 'algoliasearch-helper';
 import { InstantSearch } from './instantsearch';
 
-export type InitOptions = {
-  instantSearchInstance: InstantSearch;
-  parent: Index | null;
-  uiState: UiState;
-  state: SearchParameters;
-  helper: Helper;
-  templatesConfig: object;
-  createURL(state: SearchParameters): string;
-};
-
 export type ScopedResult = {
   indexId: string;
   results: SearchResults;
   helper: Helper;
 };
 
-export type RenderOptions = {
+type SharedRenderOptions = {
   instantSearchInstance: InstantSearch;
+  parent: Index | null;
   templatesConfig: object;
-  results: SearchResults;
   scopedResults: ScopedResult[];
   state: SearchParameters;
+  renderState: IndexRenderState;
   helper: Helper;
   searchMetadata: {
     isSearchStalled: boolean;
   };
   createURL(state: SearchParameters): string;
 };
 
+export type InitOptions = SharedRenderOptions & {
+  uiState: UiState;
```

Comment:
should we call render with uiState too? Seems simpler

Reply:
I didn't because we made the mistake to call this `uiState` instead of `initialUiState`, which would be extremely misleading when passed to `render()`. You would expect this `uiState` to be up-to-date.

Reply:
You're right, I forgot this was only initial